kernelci/runtime/lava.py

```diff
@@ -164,15 +164,28 @@ def get_results(self):
                 results[suite_name] = self._get_suite_results(tests)
         return results
 
-    def _get_stage_result(self, suite_name):
-        lava_yaml = self._data['results']['lava']
-        lava = yaml.safe_load(lava_yaml)
-        stages = {stage['name']: stage for stage in lava}
+    def _get_test_suite_result(self, child_nodes):
+        """ Calculate test suite result
+        If all child nodes pass, the suite will be marked 'pass'
+        If one of the child node fails, the suite will be marked `fail`
+        If all child nodes skipped, the suite will be marked 'skip'
+        If atleast one of the child passes and other are skipped, the suite
+        will be marked as `pass`
+        If atleast one of the child skipped and other are unknown, the
+        suite will be marked as `skip`
+        """
         result = None
-        for stage_name, stage_results in stages.items():
-            stage_name = stage_name.partition("_")[2]
-            if stage_name == suite_name:
-                result = stage_results['result']
+        result_list = [child['node']['result'] for child in child_nodes]
+        if 'fail' in result_list:
+            result = 'fail'
+        elif all(result == 'pass' for result in result_list):
+            result = 'pass'
+        elif all(result == 'skip' for result in result_list):
+            result = 'skip'
+        elif 'pass' in result_list:
+            result = 'pass'
+        elif 'skip' in result_list:
+            result = 'skip'
```

Comment:
I'd suggest simplifying this condition: `elif all(result == '$result' for result in result_list) or '$result' in result_list` leaving the current order of conditions.

This way if 1st condition is not met there's only `incomplete/skip/pass`, 2nd would exclude `pass`es and 3rd would check if there were any `skip`s. There still would be the edge case were all the results could be none of these kinds and the returned result from function would be `None` - would that be a valid value?

Reply:
> I'd suggest simplifying this condition: elif all(result == '$result' for result in result_list) or '$result' in result_list leaving the current order of conditions (fail -> pass -> skip even though as a rule I'd suggest leaving pass at the very end).

Thanks for the suggestion. I'll update it accordingly.

> There still would be the edge case were all the results could be none of these kinds and the returned result from function would be None - would that be a valid value?

Yes, `None` will be stored as `result=null` and that's valid as the field is optional.